chainregistry.go

```diff
@@ -298,16 +298,34 @@ func newChainControlFromConfig(cfg *config, chanDB *channeldb.DB,
 			if err != nil {
 				return nil, nil, err
 			}
+			// We have absolutely no idea why this this is here however this will change
+			// ports for different networks by 2. This only occurs for bitcoind/litecoind.
+			//
+			// ex: 18444 will become 18442
 			rpcPort -= 2
-			bitcoindHost = fmt.Sprintf("%v:%d",
-				bitcoindMode.RPCHost, rpcPort)
-			if cfg.Bitcoin.Active && cfg.Bitcoin.RegTest {
+
+			// Set the bitcoindHost here so that we can test if the connection works
+			// in the case of regtest. If the connection is bad then we will attempt
+			// to use a different port for the host
+			bitcoindHost = fmt.Sprintf("%v:%d", bitcoindMode.RPCHost, rpcPort)
+
+			if (cfg.Bitcoin.Active && cfg.Bitcoin.RegTest) ||
+				(cfg.Litecoin.Active && cfg.Litecoin.RegTest) {
+
 				conn, err := net.Dial("tcp", bitcoindHost)
 				if err != nil || conn == nil {
-					rpcPort = 18443
-					bitcoindHost = fmt.Sprintf("%v:%d",
-						bitcoindMode.RPCHost,
-						rpcPort)
+
+					// For Regtest we will set the ports in accordance to the chainparams for
+					// the specified network
+					if cfg.Bitcoin.Active && cfg.Bitcoin.RegTest {
+						rpcPort = 18443
```

Comment:
it seems like you wouldn't need this if `activeNetParams.rpcPort` was `18445` and `19445` for Bitcoin and Litecoin, respectively. Why aren't they set to those parameters? What assumption is LND making that is bad here?

Reply:
the whole `minus by 2` thing is super weird. All ports in LND are mapped to LTCD or BTCD values which _may not_ be the same as litecoind or bitcoind values (regtest being the example)

Reply:
So this is a problem of ltcd vs litecoind and btcd vs bitcoind (at least for regtest)?